libfwupdplugin/fu-device.c

```diff
@@ -3840,15 +3855,25 @@ fu_device_get_backend_parent_with_subsystem(FuDevice *self, const gchar *subsyst
 		const gchar *gtype_str;
 		const gchar *id;
 
-		/* we have to propagate this to preserve compat with older emulation files */
-		event = fu_device_load_event(FU_DEVICE(self), event_id, NULL);
-		if (event == NULL) {
-			g_debug("falling back to simulated device for old emulation");
-			parent =
-			    fu_backend_get_device_parent(priv->backend, self, subsystem, error);
-			if (parent != self)
-				fu_device_set_target(parent, self);
-			return g_steal_pointer(&parent);
+		/* we might have to propagate this to preserve compat with older emulation files */
+		if (fu_device_is_created_after(self, "2025-04-08T00:00:00Z")) {
```

Comment:
isn't it tied to the version of fwupd it is created from not the date it's created from?  Someone might create emulation on an older fwupd but a newer date.  Could we instead add a new field into the emulation to indicate version?

Reply:
The created attribute is from this:

      "Created" : "2025-04-09T12:21:36.246385Z",
      "GType" : "FuUdevDevice",

i.e. it's essentially a "when was this recorded" -- which I think is even better than a "recorded version".

Reply:
But I mean a field to indicate what _version of fwupd_ recorded it.  It could help us in the future if we have to special case bugs.

Reply:
I think you're completely right; if we add emulations to the stable branch we want to key it on the fwupd version, not the date the emulation was recorded. Will respin.